src/core/qgsvectorlayer.cpp

```diff
@@ -1810,6 +1810,35 @@ QString QgsVectorLayer::encodedSource( const QString &source, const QgsReadWrite
     // Refetch the source from the provider, because adding fields actually changes the source for this provider.
     src = dataProvider()->dataSourceUri();
   }
+  else if ( providerType() == QLatin1String( "virtual" ) )
+  {
+    QUrl urlSource = QUrl::fromEncoded( src.toLatin1() );
```

Comment:
Is this always latin1?

Reply:
Yes, I believe so, because `src` is the result of a [`toEncoded()`](https://doc.qt.io/qt-5/qurl.html#toEncoded) call on [a `QUrl` object](https://github.com/jdugge/QGIS/blob/virtual_layers_relative_paths/src/core/qgsvirtuallayerdefinition.cpp#L228), which returns a string in which "all non-ASCII characters are (...) percent encoded".

Reply:
@m-kuhn: Is there a more robust or more preferred way to handle the strings in this case?